membership-attribute-service/app/services/AttributesMaker.scala

```diff
@@ -40,6 +43,27 @@ class AttributesMaker extends LoggingWithLogstashFields{
 
     val groupedSubs: Map[Option[Product], List[Subscription[AnyPlan]]] = subs.groupBy(getTopProduct)
 
+    val sortedProductData: List[ProductData] = {
+      val accountsWithNonEmptySubs = subsWithAccounts.collect { case AccountWithSubscription(account, Some(subscription)) => (account, subscription) }
+
+      val groupedByProduct = accountsWithNonEmptySubs.groupBy { case (account, subscription) => getTopProduct(subscription) }
+
+      val firstSubPerProduct = groupedByProduct.collect { case (Some(k), sub :: _) => (k, sub) }
+
+      firstSubPerProduct.map { case (product, (account, subscription)) => ProductData(product, account, subscription) }.toList.sortWith(_.product.name < _.product.name)
```

Comment:
may be more readable to put the case statements on each of these collect/group onto a separate line
```
... {
 case ....
}
```

Reply:
Ok I'll do that, but I think it is especially hard to read here because Github restricts the width of the code to make sure it looks ok in computers from 1994

Reply:
🤣